iml-wire-types/wire-types.rs

```diff
@@ -1998,6 +1998,42 @@ pub struct ComponentState<T: Default> {
     pub state: T,
 }
 
+/// standard:provider:ocftype (e.g. ocf:heartbeat:ZFS, or stonith:fence_ipmilan)
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize)]
+pub struct ResourceAgentType {
+    pub standard: String,         // e.g. ocf, lsb, stonith, etc..
+    pub provider: Option<String>, // e.g. heartbeat, lustre, chroma
+    pub ocftype: String,          // e.g. Lustre, ZFS
+}
+
+impl ResourceAgentType {
+    pub fn new(standard: String, provider: Option<String>, ocftype: String) -> Self {
+        ResourceAgentType {
+            standard,
+            provider,
+            ocftype,
+        }
+    }
+}
+
+impl std::fmt::Display for ResourceAgentType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match &self.provider {
+            Some(provider) => write!(f, "{}:{}:{}", self.standard, provider, self.ocftype),
+            None => write!(f, "{}:{}", self.standard, self.ocftype),
```

Comment:
Out of curiosity: where is this used? I think looking at `foo:bar` you can't decide whether `bar` is provider or ocftype without looking ahead.

Reply:
That display format is standard for pacemaker output.